src/avt/Database/Ghost/avtUnstructuredDomainBoundaries.C

```diff
@@ -990,98 +1164,46 @@ avtUnstructuredDomainBoundaries::ExchangeMixedMaterials(vector<int> domainNum,
         //
         int matlist_cnt = oldMat->GetNZones();
         int mixlen_cnt  = oldMat->GetMixlen();
-        for (int j = 0 ; j < nTotalDomains ; j++)
+        for (int sendDom = 0 ; sendDom < nTotalDomains ; sendDom ++)
         {
-            int ngc = nGainedCells[j][domainNum[i]];
+            // create references for the domain data here
+            MixedMaterialDomainData &currDomainData = domaindata[sendDom][recvDom];
+            const int                &nGainedCells  = currDomainData.nGainedCells;
+            const std::vector<int>   &gainedMatlist = currDomainData.gainedMatlist;
+            const std::vector<int>   &gainedMixmat  = currDomainData.gainedMixmat;
+            const std::vector<float> &gainedMixvf   = currDomainData.gainedMixvf;
+
             int lml = 0; // "local" mixlen ... mixlen counter for this domain
-            for (int k = 0 ; k < ngc ; k++)
+            for (int mixedCellId = 0; mixedCellId < nGainedCells; mixedCellId ++)
```

Comment:
same issue here. The loop variable is an index not an id.

Reply:
Fixed.